tests/unit/source_code/test_notebook_linter.py

```diff
@@ -1,39 +1,174 @@
-# Place holder for notebook linter testing
-from unittest.mock import create_autospec
+import pytest
 
+from databricks.labs.ucx.hive_metastore.migration_status import MigrationIndex
 from databricks.labs.ucx.source_code import languages
+from databricks.labs.ucx.source_code.base import Deprecation, Advisory
 from databricks.labs.ucx.source_code.languages import Languages
-from databricks.labs.ucx.source_code.notebook import Notebook
 from databricks.labs.ucx.source_code.notebook_linter import NotebookLinter
 
+index = MigrationIndex([])
 
-def test_notebook_linter():
-    notebook_code = """-- Databricks notebook source
+
+@pytest.mark.parametrize(
+    "lang, source, expected, adjusted",
```

Comment:
```suggestion
    "lang, source, expected",
```
Yield already adjusted advice

Reply:
OK - originally I was asked to do this. It is a lot easier to yiled with adjustements done before

Reply:
Changed to yield only adjusted advisory